src/entities/activity/lib/services/AnswersUploadService.ts

```diff
@@ -116,7 +116,13 @@ class AnswersUploadService implements IAnswersUploadService {
     logAnswerIndex: number,
     appletId: string,
   ): Promise<string> {
-    const localFileExists = await FileSystem.exists(mediaFile.uri);
+    /** Since the application directory changes upon app update
+     * it is required to build URIs from scratch using file names
+     * just before we actually want to fetch those files.
+     * Please see: https://github.com/joltup/rn-fetch-blob/issues/204#issuecomment-786321861
+     */
+    const localFileUri = getLocalFileUri(mediaFile.fileName);
```

Comment:
Let's remove url property from MediaFile and check the code thoroughly by tsc.

Reply:
Please, ensure the change is valid for the rest of items: proto, audio / video record.

Reply:
@anq83
I agree with you on the renaming.
Regarding moving the comment, in my opinion, it is better to keep the comment in the same place as it is now because here we describe the **intent** to use the util function in the scope of the `processFileUpload` method that implements the business logic.

Reply:
The service where you placed it is complex enough. Let's do not complicate it event more by adding the comment 4-5 rows. Moreover - comments tend to become outdated and the service tends to change from time to time.
To analyze why the function has been applied - developer just clicks on F12.